api/src/main/java/io/strimzi/api/kafka/model/KafkaResources.java

```diff
@@ -0,0 +1,187 @@
+    }
+
+    /**
+     * Returns the socket address (<em>&lt;host&gt;</em>:<em>&lt;port&gt;</em>)
+     * of the internal plain bootstrap {@code Service} for a {@code Kafka} cluster of the given name.
+     * @param clusterName  The {@code metadata.name} of the {@code Kafka} resource.
+     * @return The socket address of the corresponding bootstrap {@code Service}.
+     * @see #internalTlsBootstrapConnection(String)
+     */
+    public static String internalPlainBootstrapConnection(String clusterName) {
+        return internalBootstrapServiceName(clusterName) + ":9092";
+    }
+
+    /**
+     * Returns the socket address (<em>&lt;host&gt;</em>:<em>&lt;port&gt;</em>)
+     * of the internal TLS bootstrap {@code Service} for a {@code Kafka} cluster of the given name.
+     * @param clusterName  The {@code metadata.name} of the {@code Kafka} resource.
+     * @return The socket address of the corresponding bootstrap {@code Service}.
+     * @see #internalPlainBootstrapConnection(String)
+     */
+    public static String internalTlsBootstrapConnection(String clusterName) {
+        return internalBootstrapServiceName(clusterName) + ":9093";
+    }
+
+    /**
+     * Returns the name of the (headless) brokers {@code Service} for a {@code Kafka} cluster of the given name.
+     * @param clusterName  The {@code metadata.name} of the {@code Kafka} resource.
+     * @return The name of the corresponding brokers {@code Service}.
+     */
+    public static String brokersServiceName(String clusterName) {
+        return clusterName + "-kafka-brokers";
+    }
+
+    public static String brokerConnection(String clusterName) {
+        return brokersServiceName(clusterName) + ":9092";
+    }
+
+    /**
+     * Returns the name of the external bootstrap {@code Service} for a {@code Kafka} cluster of the given name.
+     * This {@code Service} will only exist if {@code Kafka.spec.kafka.listeners.external} is configured for a
+     * loadbalancer or NodePort in the {@code Kafka} resource with the given name.
+     * @param clusterName  The {@code metadata.name} of the {@code Kafka} resource.
+     * @return The name of the corresponding bootstrap {@code Service}.
+     */
+    public static String externalBootstrapServiceName(String clusterName) {
+        return clusterName + "-kafka-external-bootstrap";
+    }
+
+    /**
+     * Returns the socket address (<em>&lt;host&gt;</em>:<em>&lt;port&gt;</em>)
+     * of the external bootstrap {@code Service} for a {@code Kafka} cluster of the given name.
+     * This {@code Service} will only exist if {@code Kafka.spec.kafka.listeners.external} is configured for a
+     * loadbalancer or NodePort in the {@code Kafka} resource with the given name.
+     * @param clusterName  The {@code metadata.name} of the {@code Kafka} resource.
+     * @return The socket address of the corresponding bootstrap {@code Service}.
+     */
+    public static String externalBootstrapConnection(String clusterName) {
```

Comment:
I do not think this actually works. You cannot connect by the service name. You should remove it.